### PR TITLE
feat: display automatically created cases & events

### PR DESCRIPTION
**Pull request by Pascal-Delange**
Goes with https://github.com/checkmarble/marble-backend/pull/495

Capture d’écran 2024-03-04 à 15 04 24

notice:
- no contributors
- no name on the events

### CODE REVIEW

**Review by balzdur — APPROVED** [review]
LGTM, implementation is 💯. I just have remarks on type declaration.

File: packages/marble-api/scripts/openapi.yaml [hunk 1]
```diff
@@ -3391,6 +3391,7 @@ components:
         propertyName: event_type
         mapping:
           case_created: '#/components/schemas/CaseCreatedEvent'
+          case_created_automatically: '#/components/schemas/CaseCreatedAutomaticallyEvent'
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply packages/marble-api/scripts/openapi.yaml:3394]: Alright, works for me 👍  I'll come back to this tomorrow. Compatible with the suggested change of "nullable userId" on the backend

File: packages/app-builder/src/components/Cases/CaseEvents.tsx [hunk 2]
```diff
@@ -312,6 +338,9 @@ export function getEventDetail(event: CaseEvent) {
     }
     case 'decision_added':
     case 'file_added': {
+      if (event.user_id === '') {
+        return <ByWorkflow type="added_by_workflow" />;
+      }
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply packages/app-builder/src/components/Cases/CaseEvents.tsx:343]: 👍  agreed

File: packages/marble-api/scripts/openapi.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply packages/marble-api/scripts/openapi.yaml]: I went with "user_id added on the different events" and "user_id is nullable on case created, decision added events"

File: packages/app-builder/src/components/Cases/CaseEvents.tsx [hunk 3]
```diff
@@ -310,8 +329,16 @@ export function getEventDetail(event: CaseEvent) {
...
+    case 'decision_added': {
+      if (!event.user_id) {
+        return <ByWorkflow type="added_by_workflow" />;
+      }
+      return <Author userId={event.user_id} type="added_by" />;
+    }
     case 'file_added': {
+      if (event.user_id === '') {
+        return <ByWorkflow type="added_by_workflow" />;
+      }
```
Comment (balzdur) [inline packages/app-builder/src/components/Cases/CaseEvents.tsx:341]: Still `''` check here, probably to remove for this event

**Review by balzdur — APPROVED** [review]

File: packages/app-builder/src/components/Cases/CaseEvents.tsx (hunk 3 above)
Comment (Pascal-Delange) [reply packages/app-builder/src/components/Cases/CaseEvents.tsx:341]: oopsies, missed that one while shifting lines

File: packages/marble-api/src/generated/marble-api.ts [hunk 4]
```diff
@@ -109,46 +109,55 @@ export type CreateCaseBody = {
...
-    user_id: string;
     created_at: string;
     event_type: string;
 };
 export type CaseCreatedEvent = {
     event_type: "case_created";
-} & CaseEventDtoBase;
+} & CaseEventDtoBase & {
+    user_id: string;
+};
```
Comment (balzdur) [inline packages/marble-api/src/generated/marble-api.ts:119]: After reading the backend PR, I realized you deleted specific events for "wrokflow" and merged them inside existing ones.
So I think `user_id` should be optionnal here

File: packages/app-builder/src/components/Cases/CaseEvents.tsx [hunk 5]
```diff
@@ -298,10 +301,26 @@ function Author({
...
+  );
+}
+
 export function getEventDetail(event: CaseEvent) {
   const { event_type } = event;
   switch (event_type) {
     case 'case_created': {
+      if (!event.user_id) {
+        return <ByWorkflow type="added_by_workflow" />;
+      }
```
Comment (balzdur) [inline packages/app-builder/src/components/Cases/CaseEvents.tsx:323]: With the `optionnal` thing, removing this line should raise a TS error here

**Review by balzdur — APPROVED** [review]

File: packages/marble-api/src/generated/marble-api.ts (hunk 4 above)
Comment (Pascal-Delange) [reply packages/marble-api/src/generated/marble-api.ts:119]: ok my bad, it's up to date in the openapi file but I must have missed a round of api generation

File: packages/app-builder/src/components/Cases/CaseEvents.tsx (hunk 5 above)
Comment (Pascal-Delange) [reply packages/app-builder/src/components/Cases/CaseEvents.tsx:323]: now does 👍

File: packages/marble-api/src/generated/marble-api.ts [hunk 6]
```diff
@@ -115,7 +115,7 @@ export type CaseEventDtoBase = {
 export type CaseCreatedEvent = {
     event_type: "case_created";
 } & CaseEventDtoBase & {
-    user_id: string;
+    user_id?: string;
```
[the remark this answers is not in the document]
Comment (Pascal-Delange) [reply packages/marble-api/src/generated/marble-api.ts:118]: I know I know, the openapi was already good, I just missed a step of regenerating the generated api file :)

**Pull request merged**